src/SaveObject.tsx

```diff
@@ -657,6 +657,9 @@ function evaluateVersionCompatibility(ver: string | number): void {
     // Prior to v2.2.0, sleeve shock was 0 to 100 internally but displayed as 100 to 0. This unifies them as 100 to 0.
     for (const sleeve of Player.sleeves) sleeve.shock = 100 - sleeve.shock;
   }
+  if (ver < 31) {
+    anyPlayer.hashManager.upgrades["Company Favor"] = anyPlayer.hashManager.upgrades["Company Favor"] ?? 0;
```

Comment:
This will error out when loading a player that doesn't have a hash manager ("Can't access property `upgrades` on `undefined` (or null, whatever the does-not-have value is)"). I imagine the game only initializes a hashmanager for players who have SF9/in BN9 but I haven't dug into that and I'm not at my regular PC to check easily.

I would do something like:

```ts
const hashUpgrades = anyPlayer.hashManager?.upgrades;
if (hashUpgrades) hashUpgrades["CompanyFavor"] ??= 0;
```

Reply:
You somehow reviewed an old version, I'd caught the error potential and had already submitted a patch for it. Thanks for the reminder that ??= exists, changed that line to use it.